In [189]:
import pandas as pd
import numpy as np

In [190]:
df = pd.read_csv(r"C:\Users\Lord\Downloads\Ecommerce Purchases")

In [191]:
df

Address    Lot AM or PM  \
0     16629 Pace Camp Apt. 448\nAlexisborough, NE 77...  46 in       PM   
1     9374 Jasmine Spurs Suite 508\nSouth John, TN 8...  28 rn       PM   
2                      Unit 0065 Box 5052\nDPO AP 27450  94 vE       PM   
3                 7780 Julia Fords\nNew Stacy, WA 45798  36 vm       PM   
4     23012 Munoz Drive Suite 337\nNew Cynthia, TX 5...  20 IE       AM   
...                                                 ...    ...      ...   
9995      966 Castaneda Locks\nWest Juliafurt, CO 96415  92 XI       PM   
9996  832 Curtis Dam Suite 785\nNorth Edwardburgh, T...  41 JY       AM   
9997              Unit 4434 Box 6343\nDPO AE 28026-0283  74 Zh       AM   
9998               0096 English Rest\nRoystad, IA 12457  74 cL       PM   
9999     40674 Barrett Stravenue\nGrimesville, WI 79682  64 Hr       AM   

                                           Browser Info  \
0     Opera/9.56.(X11; Linux x86_64; sl-SI) Presto/2...   
1     Opera/8.93.(Windows 98; Win 9x 4.90; en-US) Pr...   
2     Mozilla/5.0 (compatible; MSIE 9.0; Windows NT ...   
3     Mozilla/5.0 (Macintosh; Intel Mac OS X 10_8_0 ...   
4     Opera/9.58.(X11; Linux x86_64; it-IT) Presto/2...   
...                                                 ...   
9995  Mozilla/5.0 (Windows NT 5.1) AppleWebKit/5352 ...   
9996  Mozilla/5.0 (compatible; MSIE 9.0; Windows NT ...   
9997  Mozilla/5.0 (Macintosh; U; Intel Mac OS X 10_7...   
9998  Mozilla/5.0 (Macintosh; Intel Mac OS X 10_8_8;...   
9999  Mozilla/5.0 (X11; Linux i686; rv:1.9.5.20) Gec...   

                              Company       Credit Card CC Exp Date  \
0                     Martinez-Herman  6011929061123406       02/20   
1     Fletcher, Richards and Whitaker  3337758169645356       11/18   
2          Simpson, Williams and Pham      675957666125       08/19   
3     Williams, Marshall and Buchanan  6011578504430710       02/24   
4           Brown, Watson and Andrews  6011456623207998       10/25   
...                               ...               ...         ...   
9995                    Randall-Sloan   342945015358701       03/22   
9996         Hale, Collins and Wilson   210033169205009       07/25   
9997                     Anderson Ltd  6011539787356311       05/21   
9998                         Cook Inc   180003348082930       11/17   
9999                       Greene Inc  4139972901927273       02/19   

      CC Security Code                  CC Provider  \
0                  900                 JCB 16 digit   
1                  561                   Mastercard   
2                  699                 JCB 16 digit   
3                  384                     Discover   
4                  678  Diners Club / Carte Blanche   
...                ...                          ...   
9995               838                 JCB 15 digit   
9996               207                 JCB 16 digit   
9997                 1                VISA 16 digit   
9998               987             American Express   
9999               302                 JCB 15 digit   

                               Email                                     Job  \
0                  pdunlap@yahoo.com  Scientist, product/process development   
1                 anthony41@reed.com                       Drilling engineer   
2     amymiller@morales-harrison.com                Customer service manager   
3        brent16@olson-robinson.info                       Drilling engineer   
4        christopherwright@gmail.com                             Fine artist   
...                              ...                                     ...   
9995          iscott@wade-garner.com                              Printmaker   
9996              mary85@hotmail.com                         Energy engineer   
9997               tyler16@gmail.com                      Veterinary surgeon   
9998         elizabethmoore@reid.net                Local government officer   
9999           rachelford@vaughn.com        

In [192]:
df.head(4)

Address    Lot AM or PM  \
0  16629 Pace Camp Apt. 448\nAlexisborough, NE 77...  46 in       PM   
1  9374 Jasmine Spurs Suite 508\nSouth John, TN 8...  28 rn       PM   
2                   Unit 0065 Box 5052\nDPO AP 27450  94 vE       PM   
3              7780 Julia Fords\nNew Stacy, WA 45798  36 vm       PM   

                                        Browser Info  \
0  Opera/9.56.(X11; Linux x86_64; sl-SI) Presto/2...   
1  Opera/8.93.(Windows 98; Win 9x 4.90; en-US) Pr...   
2  Mozilla/5.0 (compatible; MSIE 9.0; Windows NT ...   
3  Mozilla/5.0 (Macintosh; Intel Mac OS X 10_8_0 ...   

                           Company       Credit Card CC Exp Date  \
0                  Martinez-Herman  6011929061123406       02/20   
1  Fletcher, Richards and Whitaker  3337758169645356       11/18   
2       Simpson, Williams and Pham      675957666125       08/19   
3  Williams, Marshall and Buchanan  6011578504430710       02/24   

   CC Security Code   CC Provider                           Email  \
0               900  JCB 16 digit               pdunlap@yahoo.com   
1               561    Mastercard              anthony41@reed.com   
2               699  JCB 16 digit  amymiller@morales-harrison.com   
3               384      Discover     brent16@olson-robinson.info   

                                      Job       IP Address Language  \
0  Scientist, product/process development  149.146.147.205       el   
1                       Drilling engineer     15.160.41.51       fr   
2                Customer service manager   132.207.160.22       de   
3                       Drilling engineer     30.250.74.19       es   

   Purchase Price  
0           98.14  
1           70.73  
2            0.95  
3           78.04

In [193]:
df.tail(4)

Address    Lot AM or PM  \
9996  832 Curtis Dam Suite 785\nNorth Edwardburgh, T...  41 JY       AM   
9997              Unit 4434 Box 6343\nDPO AE 28026-0283  74 Zh       AM   
9998               0096 English Rest\nRoystad, IA 12457  74 cL       PM   
9999     40674 Barrett Stravenue\nGrimesville, WI 79682  64 Hr       AM   

                                           Browser Info  \
9996  Mozilla/5.0 (compatible; MSIE 9.0; Windows NT ...   
9997  Mozilla/5.0 (Macintosh; U; Intel Mac OS X 10_7...   
9998  Mozilla/5.0 (Macintosh; Intel Mac OS X 10_8_8;...   
9999  Mozilla/5.0 (X11; Linux i686; rv:1.9.5.20) Gec...   

                       Company       Credit Card CC Exp Date  \
9996  Hale, Collins and Wilson   210033169205009       07/25   
9997              Anderson Ltd  6011539787356311       05/21   
9998                  Cook Inc   180003348082930       11/17   
9999                Greene Inc  4139972901927273       02/19   

      CC Security Code       CC Provider                    Email  \
9996               207      JCB 16 digit       mary85@hotmail.com   
9997                 1     VISA 16 digit        tyler16@gmail.com   
9998               987  American Express  elizabethmoore@reid.net   
9999               302      JCB 15 digit    rachelford@vaughn.com   

                           Job       IP Address Language  Purchase Price  
9996           Energy engineer   121.133.168.51       pt           25.63  
9997        Veterinary surgeon    156.210.0.254       el           83.98  
9998  Local government officer     55.78.26.143       es           38.84  
9999    Embryologist, clinical  176.119.198.199       el           67.59

In [194]:
df.dtypes

Address              object
Lot                  object
AM or PM             object
Browser Info         object
Company              object
Credit Card           int64
CC Exp Date          object
CC Security Code      int64
CC Provider          object
Email                object
Job                  object
IP Address           object
Language             object
Purchase Price      float64
dtype: object

In [195]:
df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 10000 entries, 0 to 9999
Data columns (total 14 columns):
 #   Column            Non-Null Count  Dtype  
---  ------            --------------  -----  
 0   Address           10000 non-null  object 
 1   Lot               10000 non-null  object 
 2   AM or PM          10000 non-null  object 
 3   Browser Info      10000 non-null  object 
 4   Company           10000 non-null  object 
 5   Credit Card       10000 non-null  int64  
 6   CC Exp Date       10000 non-null  object 
 7   CC Security Code  10000 non-null  int64  
 8   CC Provider       10000 non-null  object 
 9   Email             10000 non-null  object 
 10  Job               10000 non-null  object 
 11  IP Address        10000 non-null  object 
 12  Language          10000 non-null  object 
 13  Purchase Price    10000 non-null  float64
dtypes: float64(1), int64(2), object(11)
memory usage: 1.1+ MB


In [196]:
df.isnull().sum()

Address             0
Lot                 0
AM or PM            0
Browser Info        0
Company             0
Credit Card         0
CC Exp Date         0
CC Security Code    0
CC Provider         0
Email               0
Job                 0
IP Address          0
Language            0
Purchase Price      0
dtype: int64

Rows And Columns

In [197]:
df.shape

(10000, 14)

There are 10000 Rows And 14 Columns

In [198]:
df.size

140000

Highest , Lowest  And Average Purchase Prices

In [199]:
df.columns

Index(['Address', 'Lot', 'AM or PM', 'Browser Info', 'Company', 'Credit Card',
       'CC Exp Date', 'CC Security Code', 'CC Provider', 'Email', 'Job',
       'IP Address', 'Language', 'Purchase Price'],
      dtype='object')

In [200]:
df["Purchase Price"].max()

99.99

In [201]:
df["Purchase Price"].min()

0.0

In [202]:
df["Purchase Price"].mean()

50.347302

How Many People have french in their Language

In [203]:
df[df["Language"]=="fr"].count()

Address             1097
Lot                 1097
AM or PM            1097
Browser Info        1097
Company             1097
Credit Card         1097
CC Exp Date         1097
CC Security Code    1097
CC Provider         1097
Email               1097
Job                 1097
IP Address          1097
Language            1097
Purchase Price      1097
dtype: int64

Job Title Which Contain Engineer

In [204]:
df[df["Job"].str.contains("Engineer",case=False)]

Address    Lot AM or PM  \
1     9374 Jasmine Spurs Suite 508\nSouth John, TN 8...  28 rn       PM   
3                 7780 Julia Fords\nNew Stacy, WA 45798  36 vm       PM   
50      41159 Michael Centers\nAdamsfort, RI 37108-6674  46 Ce       PM   
55    27635 Maureen Bypass Apt. 883\nSandraview, SD ...  59 LJ       AM   
60    7126 Katherine Squares\nPerkinsview, CO 97299-...  63 qu       AM   
...                                                 ...    ...      ...   
9948  95544 Johnson Isle Suite 939\nMichaelberg, RI ...  91 bW       AM   
9952            9991 Vaughn Hills\nRacheltown, PA 55409  36 KC       PM   
9970              0060 Keith Stream\nWestport, CO 47097  11 nt       PM   
9977  02182 Keith Expressway\nEast Shannon, CT 20578...  34 RL       AM   
9996  832 Curtis Dam Suite 785\nNorth Edwardburgh, T...  41 JY       AM   

                                           Browser Info  \
1     Opera/8.93.(Windows 98; Win 9x 4.90; en-US) Pr...   
3     Mozilla/5.0 (Macintosh; Intel Mac OS X 10_8_0 ...   
50    Mozilla/5.0 (Windows 98; Win 9x 4.90; sl-SI; r...   
55    Mozilla/5.0 (iPod; U; CPU iPhone OS 3_3 like M...   
60    Opera/8.68.(X11; Linux x86_64; en-US) Presto/2...   
...                                                 ...   
9948  Opera/8.36.(X11; Linux x86_64; sl-SI) Presto/2...   
9952  Mozilla/5.0 (X11; Linux i686; rv:1.9.5.20) Gec...   
9970  Mozilla/5.0 (Macintosh; U; Intel Mac OS X 10_8...   
9977  Mozilla/5.0 (compatible; MSIE 9.0; Windows NT ...   
9996  Mozilla/5.0 (compatible; MSIE 9.0; Windows NT ...   

                              Company       Credit Card CC Exp Date  \
1     Fletcher, Richards and Whitaker  3337758169645356       11/18   
3     Williams, Marshall and Buchanan  6011578504430710       02/24   
50        Wright, Williams and Mendez  4008586485908075       05/19   
55                         Sims-Lyons  3158113629128344       09/19   
60                 Marshall-Fernandez   349767747049645       05/20   
...                               ...               ...         ...   
9948                     Fox-Peterson     4762924304307       03/17   
9952         Ward, Smith and Castillo  6011679271321726       09/19   
9970         Carpenter, Good and Hart  6011485664704662       07/19   
9977      Deleon, Jacobson and Benton  4186094003664688       06/21   
9996         Hale, Collins and Wilson   210033169205009       07/25   

      CC Security Code    CC Provider                        Email  \
1                  561     Mastercard           anthony41@reed.com   
3                  384       Discover  brent16@olson-robinson.info   
50                 945   JCB 16 digit    susanvalentine@obrien.org   
55                 857  VISA 16 digit       adkinsarthur@yahoo.com   
60                 672   JCB 15 digit      sweeneyhannah@jones.biz   
...                ...            ...                          ...   
9948               567     Mastercard      haleybenjamin@gmail.com   
9952               964        Voyager      jonesjennifer@olson.com   
9970               543       Discover      rangelbrian@hotmail.com   
9977               397   JCB 16 digit       daltoncarter@yahoo.com   
9996               207   JCB 16 digit           mary85@hotmail.com   

                                      Job       IP Address Language  \
1                       Drilling engineer     15.160.41.51       fr   
3                       Drilling engineer     30.250.74.19       es   
50                    Mechanical engineer  213.203.143.215       de   
55    Engineer, broadcasting (operations)   227.235.89.210       pt   
60                 Engineer, agricultural  197.144.142.102       de   
...                                   ...              ...      ...   
9948                  Structural engineer    120.36.140.58       en   
9952                     Engineer, energy    116.228.12.42       es   
9970                  Electrical engineer     242.8.85.205       en   
9977                  Biomedical e

In [205]:
df[df["Job"].str.contains("Engineer",case=False)].count()


Address             984
Lot                 984
AM or PM            984
Browser Info        984
Company             984
Credit Card         984
CC Exp Date         984
CC Security Code    984
CC Provider         984
Email               984
Job                 984
IP Address          984
Language            984
Purchase Price      984
dtype: int64

Find Email of the person with the following IP Address "132.207.160.22"

In [206]:
df.columns

Index(['Address', 'Lot', 'AM or PM', 'Browser Info', 'Company', 'Credit Card',
       'CC Exp Date', 'CC Security Code', 'CC Provider', 'Email', 'Job',
       'IP Address', 'Language', 'Purchase Price'],
      dtype='object')

In [207]:
df[df["IP Address"]=="132.207.160.22"]



Address    Lot AM or PM  \
2  Unit 0065 Box 5052\nDPO AP 27450  94 vE       PM   

                                        Browser Info  \
2  Mozilla/5.0 (compatible; MSIE 9.0; Windows NT ...   

                      Company   Credit Card CC Exp Date  CC Security Code  \
2  Simpson, Williams and Pham  675957666125       08/19               699   

    CC Provider                           Email                       Job  \
2  JCB 16 digit  amymiller@morales-harrison.com  Customer service manager   

       IP Address Language  Purchase Price  
2  132.207.160.22       de            0.95

In [208]:
df[df["IP Address"]=="132.207.160.22"].count()



Address             1
Lot                 1
AM or PM            1
Browser Info        1
Company             1
Credit Card         1
CC Exp Date         1
CC Security Code    1
CC Provider         1
Email               1
Job                 1
IP Address          1
Language            1
Purchase Price      1
dtype: int64

How many people have mastercard as their Credit Card Provider And made a purchase above 50 ?

In [209]:
df.columns

Index(['Address', 'Lot', 'AM or PM', 'Browser Info', 'Company', 'Credit Card',
       'CC Exp Date', 'CC Security Code', 'CC Provider', 'Email', 'Job',
       'IP Address', 'Language', 'Purchase Price'],
      dtype='object')

In [210]:
df["CC Provider"]

0                      JCB 16 digit
1                        Mastercard
2                      JCB 16 digit
3                          Discover
4       Diners Club / Carte Blanche
                   ...             
9995                   JCB 15 digit
9996                   JCB 16 digit
9997                  VISA 16 digit
9998               American Express
9999                   JCB 15 digit
Name: CC Provider, Length: 10000, dtype: object

In [211]:
df[(df["CC Provider"] == "Mastercard")  &  (df["Purchase Price"] > 50)].count()


Address             405
Lot                 405
AM or PM            405
Browser Info        405
Company             405
Credit Card         405
CC Exp Date         405
CC Security Code    405
CC Provider         405
Email               405
Job                 405
IP Address          405
Language            405
Purchase Price      405
dtype: int64

Find Email of the person With The Following Cerdit Card Number:
4664825258997302

In [212]:
df[df["Credit Card"] == 4664825258997302]

Address    Lot AM or PM  \
9992  885 Allen Mountains Apt. 230\nWallhaven, LA 16995  40 vH       PM   

                                           Browser Info    Company  \
9992  Mozilla/5.0 (Macintosh; PPC Mac OS X 10_6_5) A...  Wells Ltd   

           Credit Card CC Exp Date  CC Security Code CC Provider  \
9992  4664825258997302       10/20               431    Discover   

                  Email           Job     IP Address Language  Purchase Price  
9992  bberry@wright.net  Set designer  174.173.51.32       de           67.96

In [213]:
df[df["Credit Card"] == 4664825258997302]["Email"]

9992    bberry@wright.net
Name: Email, dtype: object

How many people purchase During  AM and How many people purchase During PM

In [214]:
df.columns

Index(['Address', 'Lot', 'AM or PM', 'Browser Info', 'Company', 'Credit Card',
       'CC Exp Date', 'CC Security Code', 'CC Provider', 'Email', 'Job',
       'IP Address', 'Language', 'Purchase Price'],
      dtype='object')

In [215]:
df["AM or PM"].value_counts()

AM or PM
PM    5068
AM    4932
Name: count, dtype: int64

How many people have a credit card that expire in 2020 ?

In [216]:
df["CC Exp Date"].value_counts()

CC Exp Date
08/22    108
01/26    105
04/19    105
01/21    104
01/19    103
        ... 
12/22     67
11/25     67
10/25     64
08/26     48
08/16     39
Name: count, Length: 121, dtype: int64

In [217]:
def fun():
    count = 0
    for date in df["CC Exp Date"]:
        if date.split("/")[1] == "20":
            count = count +1
    print(count)

In [218]:
fun()

988


or

In [219]:
df[df["CC Exp Date"].apply(lambda x:x[3:]=="20")].count()

Address             988
Lot                 988
AM or PM            988
Browser Info        988
Company             988
Credit Card         988
CC Exp Date         988
CC Security Code    988
CC Provider         988
Email               988
Job                 988
IP Address          988
Language            988
Purchase Price      988
dtype: int64

Top 5 Email provider

In [220]:
df.columns

Index(['Address', 'Lot', 'AM or PM', 'Browser Info', 'Company', 'Credit Card',
       'CC Exp Date', 'CC Security Code', 'CC Provider', 'Email', 'Job',
       'IP Address', 'Language', 'Purchase Price'],
      dtype='object')

In [221]:
list = []

for email in df["Email"]:
    list.append(email.split("@")[1])


df["temp"]=list


df["temp"].value_counts().head()

temp
hotmail.com     1638
yahoo.com       1616
gmail.com       1605
smith.com         42
williams.com      37
Name: count, dtype: int64

Conclusion

The Highest Purchase Price are 99.99
the Lowest  Purchase Price are 0.0
And 
Average Purchase Prices are 50.347302


The people that have mastercard as their Credit Card Provider And made a purchase above 50 are 405

How many people purchase During  AM and How many people purchase During PM

PM    5068

AM    4932

Top 5 Email provider

hotmail.com     1638

yahoo.com       1616

gmail.com       1605

smith.com         42

williams.com      37